File: frontend/src/partials/services/RevenueService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/revenues';

class RevenueService {
  getAllRevenues() {
    return axios.get(API_URL);
  }

  getRevenueById(id) {
    return axios.get(`${API_URL}/${id}`);
  }

  createRevenue(revenue) {
    return axios.post(API_URL, revenue);
  }

  updateRevenue(id, revenue) {
    return axios.put(`${API_URL}/${id}`, revenue);
  }

  deleteRevenue(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
}

export default new RevenueService();

File: backend/src/main/java/com/louaysaafi/HotelManagementSystem/services/RevenueService.java
package com.louaysaafi.HotelManagementSystem.services;

import com.louaysaafi.HotelManagementSystem.models.Revenue;
import com.louaysaafi.HotelManagementSystem.repositories.RevenueRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class RevenueService {
    @Autowired
    private RevenueRepository revenueRepository;

    public List<Revenue> getAllRevenues() {
        return revenueRepository.findAll();
    }

    public Optional<Revenue> getRevenueById(Long id) {
        return revenueRepository.findById(id);
    }

    public Revenue saveRevenue(Revenue revenue) {
        return revenueRepository.save(revenue);
    }

    public Revenue updateRevenue(Long id, Revenue revenueDetails) {
        Revenue revenue = revenueRepository.findById(id).orElseThrow(() -> new RuntimeException("Revenue not found"));
        revenue.setDate(revenueDetails.getDate());
        revenue.setDirectRevenue(revenueDetails.getDirectRevenue());
        revenue.setIndirectRevenue(revenueDetails.getIndirectRevenue());
        return revenueRepository.save(revenue);
    }

    public void deleteRevenue(Long id) {
        revenueRepository.deleteById(id);
    }
}
